pkg/operator/certrotation/target_test.go

```diff
@@ -631,3 +636,217 @@ func TestEnsureTargetSignerCertKeyPair(t *testing.T) {
+	// to the lister, we simulate the race condition
+	// that causes overwriting of the CM and
+	// dropping the firstSigner
+	//
+	// nevertheless, let's add the secondSigner to our expectedCertificates
+	expectedCertificates = append(expectedCertificates, firstSigner.Config.Certs[0])
+	require.Equal(t, expectedCertificates, res)
+
+	res, err = target.EnsureConfigMapCABundle(context.TODO(), secondSigner)
+	require.NoError(t, err)
+	expectedCertificates = append(expectedCertificates, secondSigner.Config.Certs[0])
+	sort.SliceStable(expectedCertificates, func(i, j int) bool {
+		return bytes.Compare(expectedCertificates[i].Raw, expectedCertificates[j].Raw) < 0
+	})
+	require.Equal(t, expectedCertificates, res)
+}
+
+// TestCABundleWithMultipleSignersThree shows that a CABundle can miss a CAs
+func TestCABundleWithMultipleSignersThree(t *testing.T) {
+	firstSignerSecret := &corev1.Secret{}
+	require.NoError(t, setSigningCertKeyPairSecret(firstSignerSecret, time.Hour))
+	secondSignerSecret := &corev1.Secret{}
+	require.NoError(t, setSigningCertKeyPairSecret(secondSignerSecret, time.Hour))
+	thirdSignerSecret := &corev1.Secret{}
+	require.NoError(t, setSigningCertKeyPairSecret(thirdSignerSecret, time.Hour))
+
+	firstSecretCommonName := firstSignerSecret.Annotations[CertificateIssuer]
+	secondSecretCommonName := secondSignerSecret.Annotations[CertificateIssuer]
+	thirdSecretCommonName := thirdSignerSecret.Annotations[CertificateIssuer]
+	require.Equal(t, firstSecretCommonName, secondSecretCommonName, thirdSecretCommonName)
+
+	firstSigner, err := crypto.GetCAFromBytes(firstSignerSecret.Data["tls.crt"], firstSignerSecret.Data["tls.key"])
+	require.NoError(t, err)
+	secondSigner, err := crypto.GetCAFromBytes(secondSignerSecret.Data["tls.crt"], secondSignerSecret.Data["tls.key"])
+	require.NoError(t, err)
+	thirdSigner, err := crypto.GetCAFromBytes(thirdSignerSecret.Data["tls.crt"], thirdSignerSecret.Data["tls.key"])
+	require.NoError(t, err)
+
+	fakeKubeClient := kubefake.NewSimpleClientset()
+	coreV1Indexer := cache.NewIndexer(cache.MetaNamespaceKeyFunc, cache.Indexers{cache.NamespaceIndex: cache.MetaNamespaceIndexFunc})
+
+	target := CABundleConfigMap{
+		Namespace:     "ns",
+		Name:          "trust-bundle",
+		Client:        fakeKubeClient.CoreV1(),
+		Lister:        corev1listers.NewConfigMapLister(coreV1Indexer),
+		EventRecorder: events.NewInMemoryRecorder("test"),
+	}
+
+	expectedCertificates := []*x509.Certificate{firstSigner.Config.Certs[0]}
+	res, err := target.EnsureConfigMapCABundle(context.TODO(), firstSigner)
+	require.NoError(t, err)
+	require.Equal(t, expectedCertificates, res)
+	caBundleConfigMap, err := fakeKubeClient.CoreV1().ConfigMaps("ns").Get(context.Background(), "trust-bundle", metav1.GetOptions{})
+	require.NoError(t, err)
+	coreV1Indexer.Add(caBundleConfigMap)
+
+	res, err = target.EnsureConfigMapCABundle(context.TODO(), secondSigner)
+	require.NoError(t, err)
+	// by not adding the raBundleConfigMap
```

Comment:
I don't understand what would be the fix here. There is no way to fix `EnsureConfigMapCABundle` to make it pass as we intentionally ignore result of this function call